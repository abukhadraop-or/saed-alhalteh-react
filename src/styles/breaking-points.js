const breakingPoints = {
  mobile: '30rem',
  tablets: '48rem',
  smallScreen: '64.0625rem',
  largeScreen: '75rem',
};

export default breakingPoints;
