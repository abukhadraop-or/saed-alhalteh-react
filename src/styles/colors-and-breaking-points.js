export const colors = {
  black: '#000',
  blueViolet: '#4F52B0',
  concrete: '#f3f3f3',
  dustyGrey: '#999',
  edward: '#acb9b4',
  fern: '#5cb85c',
  goblin: '#347634',
  gray: '#868686',
  mineShaft: '#333333',
  oBlack: 'rgba(0,0,0,0.75)',
  o3Black: 'rgba(0,0,0,0.3)',
  oFern: 'rgba(92,184,92,0.5)',
  silver: '#c8c8c8',
  white: '#fff',
};

export const breakingPoints = {
  mobile: '480px',
  tablets: '768px',
  smallScreen: '1025px',
  largeScreen: '1200px',
};
