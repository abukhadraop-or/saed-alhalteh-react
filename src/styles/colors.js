const colors = {
  black: '#000000',
  blueViolet: '#4f52b0',
  concrete: '#f3f3f3',
  darkGreen: '#347634',
  dustyGrey: '#999999',
  gray: '#868686',
  green: '#5cb85c',
  lightGrey: '#acb9b4',
  mineShaft: '#333333',
  partiallyDarkBlack: 'rgba(0,0,0,0.75)',
  silver: '#c8c8c8',
  transparentBlack: 'rgba(0,0,0,0.3)',
  transparentGreen: 'rgba(92,184,92,0.5)',
  white: '#ffffff',
};

export default colors;
