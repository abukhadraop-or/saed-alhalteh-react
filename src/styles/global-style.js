import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
  body {
    font-family:'Titillium Web', sans-serif;
    margin: 0;
  }
`;

export default GlobalStyle;
