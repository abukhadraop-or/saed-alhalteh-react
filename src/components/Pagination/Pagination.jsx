import React from 'react';
import PropTypes from 'prop-types';
import { PaginationList, PaginationLink } from './pagination.styles';

/**
 * Component that displays the pages number based on the number of available posts.
 *
 * @param {Array} posts        Array of objects, which contains the posts.
 * @param {Number} pageSize    Maximum number of posts that are allowed per page.
 * @param {function} onClick   functions to set the range of posts to be displayed based on the current page.
 * @param {Number} CurrentPage current page.
 *
 * @returns {JSX.Element} Pagination
 */

function Pagination({ posts, pageSize, onClick, currentPage }) {
  const itemCount = posts.length;
  let pagesCount = Math.floor(itemCount / pageSize) + 1;
  if (itemCount % pageSize === 0) pagesCount -= 1;
  if (pagesCount === 1) return null;
  const pagesArray = [];
  for (let i = 1; i <= pagesCount; i += 1) {
    pagesArray.push(i);
  }
  return (
    <div>
      <PaginationList currentPage={currentPage}>
        {pagesArray.map((page) => (
          <PaginationLink key={page} onClick={() => onClick(page)}>
            {page}
          </PaginationLink>
        ))}
      </PaginationList>
    </div>
  );
}
Pagination.propTypes = {
  pageSize: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
  currentPage: PropTypes.number.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  posts: PropTypes.array.isRequired,
};
export default Pagination;
