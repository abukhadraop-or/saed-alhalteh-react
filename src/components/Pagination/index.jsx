import PropTypes from 'prop-types';
import React from 'react';
import { PaginationButton, PaginationItem, PaginationList } from './pagination.styles';

/**
 * Pagination.
 *
 * @param {Object} props The props object.
 * @param {number} props.count  Posts count.
 * @param {number} props.pageSize Maximum allowed number of posts per page.
 * @param {Function} props.onClick Handles clicks on pagination.
 * @param {Number} props.currentPage The current page.
 *
 * @return {JSX.Element} Pagination component.
 */
function Pagination({ count, pageSize, onClick, currentPage }) {
  let pagesCount = Math.floor(count / pageSize) + 1;
  pagesCount = count % pageSize === 0 ? pagesCount - 1 : pagesCount;

  const pagesArray = [];

  for (let i = 1; i <= pagesCount; i += 1) {
    pagesArray.push(i);
  }

  return pagesCount === 1 ? null : (
    <PaginationList currentPage={currentPage}>
      {pagesArray.map((page) => (
        <PaginationItem key={page}>
          <PaginationButton onClick={() => onClick(page)}>{page}</PaginationButton>
        </PaginationItem>
      ))}
    </PaginationList>
  );
}

Pagination.propTypes = {
  pageSize: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
  currentPage: PropTypes.number.isRequired,
  count: PropTypes.number.isRequired,
};

export default Pagination;
