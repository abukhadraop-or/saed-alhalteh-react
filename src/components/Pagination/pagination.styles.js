import styled from 'styled-components';
import { colors } from 'shared/styles';

export const PaginationList = styled.ul`
  list-style: none;
  text-align: center;

  & :nth-child(${(props) => props.currentPage}) {
    background-color: ${colors.fern};
    color: ${colors.white};
  }
`;

export const PaginationLink = styled.li`
  border: 0.0625rem solid ${colors.fern};
  border-radius: 0.1875rem;
  color: ${colors.fern};
  cursor: pointer;
  display: inline-block;
  margin: 0 0.3125rem 0 0;
  padding: 0.4375rem;
`;
