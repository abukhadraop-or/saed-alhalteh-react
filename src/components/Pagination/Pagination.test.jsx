import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Pagination from '.';
import { PaginationButton } from './pagination.styles';

describe('<Pagination /> tests', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(
      <Pagination pageSize={5} onClick={jest.fn()} currentPage={1} count={25} />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('displays no pagination buttons when count is less than pageSize  ', () => {
    const wrapper = shallow(
      <Pagination pageSize={10} onClick={jest.fn()} currentPage={1} count={5} />
    );

    expect(wrapper.find(PaginationButton).length).toBe(0);
  });

  it('displays no pagination buttons when count is equal to pageSize  ', () => {
    const wrapper = shallow(
      <Pagination pageSize={10} onClick={jest.fn()} currentPage={1} count={10} />
    );

    expect(wrapper.find(PaginationButton).length).toBe(0);
  });

  it('displays pagination buttons when count is larger than pageSize  ', () => {
    const wrapper = shallow(
      <Pagination pageSize={5} onClick={jest.fn()} currentPage={1} count={10} />
    );

    expect(wrapper.find(PaginationButton).length).toBe(2);
  });

  it('calls onClick function when pagination button is clicked  ', () => {
    const mock = jest.fn();
    const wrapper = shallow(<Pagination pageSize={5} onClick={mock} currentPage={1} count={10} />);

    wrapper.find(PaginationButton).at(1).simulate('click');

    expect(mock).toHaveBeenCalledWith(2);
  });
});
