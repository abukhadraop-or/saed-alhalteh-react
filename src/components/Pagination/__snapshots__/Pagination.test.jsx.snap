// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> tests matches the snapshot 1`] = `
<styled.ul
  currentPage={1}
>
  <styled.li
    key="1"
  >
    <styled.button
      onClick={[Function]}
    >
      1
    </styled.button>
  </styled.li>
  <styled.li
    key="2"
  >
    <styled.button
      onClick={[Function]}
    >
      2
    </styled.button>
  </styled.li>
  <styled.li
    key="3"
  >
    <styled.button
      onClick={[Function]}
    >
      3
    </styled.button>
  </styled.li>
  <styled.li
    key="4"
  >
    <styled.button
      onClick={[Function]}
    >
      4
    </styled.button>
  </styled.li>
  <styled.li
    key="5"
  >
    <styled.button
      onClick={[Function]}
    >
      5
    </styled.button>
  </styled.li>
</styled.ul>
`;
